tests/integration/api/error_group_callback/test_error_group_callback_empty_callback.php

```diff
@@ -0,0 +1,109 @@
+    [{"name": "DurationByCaller/Unknown/Unknown/Unknown/Unknown/allOther"}, [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Errors/OtherTransaction/php__FILE__"},                       [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Errors/all"},                                                [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Errors/allOther"},                                           [1, "??", "??", "??", "??", "??"]],
+    [{"name": "ErrorsByCaller/Unknown/Unknown/Unknown/Unknown/all"},        [1, "??", "??", "??", "??", "??"]],
+    [{"name": "ErrorsByCaller/Unknown/Unknown/Unknown/Unknown/allOther"},   [1, "??", "??", "??", "??", "??"]],
+    [{"name": "OtherTransaction/all"},                                      [1, "??", "??", "??", "??", "??"]],
+    [{"name": "OtherTransaction/php__FILE__"},                              [1, "??", "??", "??", "??", "??"]],
+    [{"name": "OtherTransactionTotalTime"},                                 [1, "??", "??", "??", "??", "??"]],
+    [{"name": "OtherTransactionTotalTime/php__FILE__"},                     [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Supportability/Logging/Forwarding/PHP/enabled"},             [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Supportability/Logging/Metrics/PHP/enabled"},                [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Supportability/api/notice_error"},                           [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Supportability/api/set_error_group_callback"},               [1, 0, 0, 0, 0, 0]]
+  ]
+]
+*/
+
+/*EXPECT_ERROR_EVENTS
+[
+  "?? agent run id",
+  {
+    "reservoir_size": 100,
+    "events_seen": 1
+  },
+  [
+    [
+      {
+        "type": "TransactionError",
+        "timestamp": "??",
+        "error.class": "Exception",
+        "error.message": "Noticed exception 'Exception' with message 'Sample Exception' in __FILE__:??",
+        "transactionName": "OtherTransaction\/php__FILE__",
+        "duration": "??",
+        "nr.transactionGuid": "??",
+        "guid": "??",
+        "sampled": true,
+        "priority": "??",
+        "traceId": "??",
+        "spanId": "??"
+      },
+      {},
+      {}
+    ]
+  ]
+]
+*/
+
+/*EXPECT_TRACED_ERRORS
+[
+  "?? agent run id",
+  [
+    [
+      "??",
+      "OtherTransaction\/php__FILE__",
+      "Noticed exception 'Exception' with message 'Sample Exception' in __FILE__:??",
+      "Exception",
+      {
+        "stack_trace": [
+          " in alpha called at __FILE__ (??)"
+        ],
+        "agentAttributes": {},
+        "intrinsics": {
+          "totalTime": "??",
+          "cpu_time": "??",
+          "cpu_user_time": "??",
+          "cpu_sys_time": "??",
+          "guid": "??",
+          "sampled": true,
+          "priority": "??",
+          "traceId": "??"
+        }
+      }
+    ]
+  ]
+]
+*/
+
+function alpha()
+{
+  newrelic_notice_error(new Exception('Sample Exception'));
+}
+
+newrelic_set_error_group_callback();
```

Comment:
Should the return value for `newrelic_set_error_group_callback` be tested? This test expects `false` to be returned.

Reply:
[`tap_assert`](https://github.com/newrelic/newrelic-php-agent/blob/main/tests/include/tap.php#L12) and [`tap_refute`](https://github.com/newrelic/newrelic-php-agent/blob/main/tests/include/tap.php#L21) with `/*EXPECT*` could be used to test return value of `newrelic_set_error_group_callback()`. 

For an example use of `tap_assert` with corresponding `/*EXPECT` see:

[`/*EXPECT ok - delete list with ltrim */`](https://github.com/newrelic/newrelic-php-agent/blob/main/tests/integration/redis/test_list.php#L47) 

[`tap_assert($redis->ltrim($key, 1, 0), 'delete list with ltrim');`](https://github.com/newrelic/newrelic-php-agent/blob/main/tests/integration/redis/test_list.php#L195)

Here it could be something like this:
```
/*EXPECT
ok - should reject zero args
*/

@$result = newrelic_set_error_group_callback();
tap_refute($result, 'should reject zero args')
```

Reply:
Good catch- I'll add a tap-ok/not ok EXPECT to these tests when calling the API

Reply:
I think this approach should help with eliminating the need for separate treatment of error handling by different PHPs in a few tests (like test_error_group_callback_bad_params.php), and use of very verbose `/*EXPECT_REGEX`.

Reply:
> I think this approach should help with eliminating the need for separate treatment of error handling by different PHPs in a few tests (like test_error_group_callback_bad_params.php), and use of very verbose /*EXPECT_REGEX.

I know the EXPECT_REGEX is pretty intense, but I really do believe we need it- without it, we don't really have the ability to test that we are passing the correct data (or any data at all) to the customer callback. WIthout it we would only be testing that the API was called successfully and *something* happened that generated an `error.group.name` Agent attribute.

Reply:
(unless you're only referring to removing the EXPECT_REGEX for the error output in `bad_callback` and `bad_callback.php7` - those can go. The others are testing the values we provide, so they should really remain IMO.)

Reply:
@bduranleau-nr You have far more context on this so it's best for you to decide if and which EXPECT_REGEX could be simplified. I'll go with your recommendation on this.

Reply:
in practice it looks like merging the bad_callback tests is not possible- for  php 8.0 it treats this as a fatal error and will not continue execution, including printing out the tap test result. This is due to `zend_parse_parameters` in the API code expecting an argument of a function type, which bad_callback intentionally does not provide. This differs from PHP 7, where it is treated as an exception and execution continues as normal.
I can remove the EXPECT_ERROR_EVENTS and EXPECT_METRICS from both of these files, but it looks like EXPECT_REGEX has to stay.

However, I did manage to add tap tests to all the other tests to verify the correct return value from the API call.

Reply:
Integration test improvements adding tap output made in af742dccca975e2841f3d47d61e30d2361f53134